tests/transformations/test_transformations.py

```diff
@@ -49,6 +49,43 @@ def copied_purchases(dataset: SupportsReadableDataset[Any]) -> Any:
     }
 
 
+@pytest.mark.parametrize(
+    "destination_config",
+    transformation_configs(only_duckdb=True),
+    ids=lambda x: x.name,
+)
+def test_transformations_with_supplied_hints(
+    destination_config: DestinationTestConfiguration,
+) -> None:
+    fruit_p, dest_p = setup_transformation_pipelines(destination_config)
+
+    s = fruitshop_source()
+    s.inventory.apply_hints(columns={"price": {"precision": 10, "scale": 2}})
+    fruit_p.run(s)
+
+    assert fruit_p.default_schema.tables["inventory"]["columns"]["price"]["precision"] == 10
+    assert fruit_p.default_schema.tables["inventory"]["columns"]["price"]["scale"] == 2
+
+    # we can now transform this table twice, one with changed hints and once with the original hints
+    @dlt.transformation()
+    def inventory_original(dataset: SupportsReadableDataset[Any]) -> Any:
+        return dataset["inventory"]
+
+    @dlt.transformation(columns={"price": {"precision": 20, "scale": 2}})
```

Comment:
I think this is the correct behavior, we should allow to set the precision for an existing column here. Or would you rather do this in the function body? I am not sure.